pylsp/plugins/jedi_completion.py

```diff
@@ -219,11 +220,21 @@ def _format_completion(d, markup_kind: str, include_params=True, resolve=False,
     if resolve:
         completion = _resolve_completion(completion, d, markup_kind)
 
+    # Adjustments for file completions
     if d.type == 'path':
         path = osp.normpath(d.name)
         path = path.replace('\\', '\\\\')
         path = path.replace('/', '\\/')
-        completion['insertText'] = path
+
+        # If the completion ends with os.sep, it means it's a directory. So we add an escaped os.sep
+        # at the end to ease additional file completions.
+        if d.name.endswith(os.sep):
+            if os.name == 'nt':
+                completion['insertText'] = path + '\\\\'
+            else:
+                completion['insertText'] = path + '\\/'
+        else:
+            completion['insertText'] = path
```

Comment:
- Would result in less repetition to mutate the `path` and have `completion['insertText'] = path` at the end like before.
- Should this code verify that the trailing slash doesn't exist before adding one? Otherwise it could lead to repeated slashes if jedi decides to change its behavior

Reply:
> Would result in less repetition to mutate the path and have completion['insertText'] = path at the end like before.

Good point, I'll do the change.

> Should this code verify that the trailing slash doesn't exist before adding one?

That's taking care of by `os.normpath` above.